crates/tiny/src/cmd.rs

```diff
@@ -450,31 +464,36 @@ fn names(args: CmdArgs) {
     } = args;
     let words: Vec<&str> = args.split_whitespace().collect();
 
-    let client = match find_client(clients, src.serv_name()) {
-        None => {
-            return;
-        }
-        Some(client) => client,
-    };
-
-    if let MsgSource::Chan { ref serv, ref chan } = src {
-        let nicks_vec = client.get_chan_nicks(chan);
-        let target = MsgTarget::Chan { serv, chan };
-        if words.is_empty() {
-            ui.add_client_msg(
-                &format!("{} users: {}", nicks_vec.len(), nicks_vec.join(", ")),
-                &target,
-            );
-        } else {
-            let nick = words[0];
-            if nicks_vec.iter().any(|v| v == nick) {
-                ui.add_client_msg(&format!("{} is online", nick), &target);
-            } else {
-                ui.add_client_msg(&format!("{} is not in the channel", nick), &target);
+    if let Some(serv_name) = src.serv_name() {
+        match find_client(clients, serv_name) {
+            None => {
+                return;
+            }
+            Some(client) => {
+                if let MsgSource::Chan { ref serv, ref chan } = src {
+                    let nicks_vec = client.get_chan_nicks(chan);
+                    let target = MsgTarget::Chan { serv, chan };
+                    if words.is_empty() {
+                        ui.add_client_msg(
+                            &format!("{} users: {}", nicks_vec.len(), nicks_vec.join(", ")),
+                            &target,
+                        );
+                    } else {
+                        let nick = words[0];
+                        if nicks_vec.iter().any(|v| v == nick) {
+                            ui.add_client_msg(&format!("{} is online", nick), &target);
+                        } else {
+                            ui.add_client_msg(&format!("{} is not in the channel", nick), &target);
+                        }
+                    }
+                } else {
+                    ui.add_client_err_msg(
+                        "/names only supported in chan tabs",
+                        &MsgTarget::CurrentTab,
+                    );
+                }
```

Comment:
This diff should be much smaller if you implement this in the style of previous version:

```rust
let serv_name = match src.serv_name() {
    None => return,
    Some(serv_name) => serv_name,
};
```

Then the lines starting after `let client = ...` should not change at all.